net4j/com.b2international.snowowl.eventbus/src/com/b2international/snowowl/internal/eventbus/EventBus.java

```diff
@@ -181,6 +211,28 @@ public IEventBus unregisterHandler(String address, IHandler<IMessage> handler) {
 		return this;
 	}
 	
+	private void decrementCounter(final String tag, final Map<String, AtomicLong> toDecrement) {
+		final AtomicLong counter = getOrCreateCounter(tag, toDecrement);
+		counter.set(counter.decrementAndGet());
+		toDecrement.put(tag, counter);
+	}
+	
+	private void incrementCounter(final String tag, final Map<String, AtomicLong> toIncrement) {
+		final AtomicLong counter = getOrCreateCounter(tag, toIncrement);
+		counter.set(counter.incrementAndGet());
+		toIncrement.put(tag, counter);
+	}
+	
+	private AtomicLong getOrCreateCounter(final String tag, final Map<String, AtomicLong> counterMap) {
+		if (counterMap.containsKey(tag)) {
+			return counterMap.get(tag);
+		} else {
+			final AtomicLong counter = new AtomicLong(0L);
```

Comment:
It is possible that two threads will execute this block, make sure that if a second thread execute the `Map.put` it will write back the original value unless it is `null`.